### PR TITLE
adding BM type in meta datafor graph output

### PR DESCRIPTION
**Pull request by vutukuriajay2241**

### CODE REVIEW

File: asctb-api/src/models/graph.model.ts [hunk 1]
```diff
@@ -44,19 +45,38 @@ export class GNode {
...
   label: string;
   name: string;
+  ontologyId: string;
+  bmType?: string;
 
-  constructor(name: string, ontologyId: string, label: string) {
-    this.name = name;
+  constructor(name: string, ontologyId: string, label: string, bmType?: string) {
     this.ontologyId = ontologyId;
+    if (ontologyId.toLowerCase().startsWith('fma')) {
```
Comment (bherr2) [inline asctb-api/src/models/graph.model.ts:62]: Why are you doing this? Ids at this point should already be in the form `${ontologyType}:${ontologyId}`...

File: asctb-api/src/functions/graph.functions.ts [hunk 2]
```diff
@@ -24,7 +25,7 @@ export function buildgraphAS(data: Row[], graphData: GraphData) {
 
     row.anatomical_structures.forEach((structure) => {
       let s: number;
-      if (structure.id) {
+      if (structure.id && structure.id.toLowerCase() !== 'not found') {
```
Comment (bherr2) [inline asctb-api/src/functions/graph.functions.ts:28]: We should probably clear out 'not found' ids using the fixOntologyId function https://github.com/hubmapconsortium/ccf-asct-reporter/blob/addingBMTypeToGraphOut/asctb-api/src/functions/lookup.functions.ts#L15

File: asctb-api/src/models/graph.model.ts (hunk 1 above)
Comment (vutukuriajay2241) [reply asctb-api/src/models/graph.model.ts:62]: I was still in the intuition that ID's are not fixed. I will do the necessary changes and will commit again

Comment (bherr2) [reply asctb-api/src/models/graph.model.ts:62]: Ok, IDS should be fixed now.

**Comment by bherr2** [comment]
@vutukuriajay2241 please try to loop back to this. It is getting merge conflicts as things evolve and will eventually be lost with time.

File: asctb-api/src/functions/graph.functions.ts [hunk 3]
```diff
@@ -42,7 +43,7 @@ export function buildgraphAS(data: Row[], graphData: GraphData) {
         id += 1;
         const newNode = new GNode(
           id,
-          structure.name,
+          structure.id.toLowerCase() !== 'not found' && structure.id && idNameSet[structure.id] ? idNameSet[structure.id]  : structure.name,
```
Comment (bherr2) [inline asctb-api/src/functions/graph.functions.ts:46]: missed a spot :grin:

File: asctb-api/src/functions/lookup.functions.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ export function buildHGNCLink(id: string): string {
 }
 
 export function fixOntologyId(id: string): string {
-  if (id?.toLowerCase() === 'n/a') {
+  if (id?.toLowerCase() === 'n/a' || id?.toLowerCase() === 'not found') {
```
Comment (bherr2) [inline asctb-api/src/functions/lookup.functions.ts:24]: Oh and another spot

**Review by bherr2 — APPROVED** [review]

**Pull request merged**